### PR TITLE
report(metrics): use css grid so metrics are aligned

### PR DESCRIPTION
**Pull request by connorjclark**
before:

image

after:

image

### CODE REVIEW

**Comment by patrickhulce** [comment]
there needs to be a 😍 reaction on PRs :)

File: lighthouse-core/report/html/renderer/performance-category-renderer.js [hunk 1]
```diff
@@ -136,15 +136,13 @@ class PerformanceCategoryRenderer extends CategoryRenderer {
     const keyMetrics = metricAudits.slice(0, 3);
     const otherMetrics = metricAudits.slice(3);
 
-    const metricsBoxesEl = this.dom.createChildOf(metricAuditsEl, 'div', 'lh-columns');
-    const metricsColumn1El = this.dom.createChildOf(metricsBoxesEl, 'div', 'lh-column');
-    const metricsColumn2El = this.dom.createChildOf(metricsBoxesEl, 'div', 'lh-column');
+    const metricsBoxesEl = this.dom.createChildOf(metricAuditsEl, 'div', 'lh-metrics-container');
 
     keyMetrics.forEach(item => {
```
Comment (paulirish) [inline lighthouse-core/report/html/renderer/performance-category-renderer.js:141]: i think can just go with `metricAudits.forEach......` and drop these two lil arrays, right? same order?

File: lighthouse-core/report/html/report-styles.css [hunk 2]
```diff
@@ -544,50 +544,33 @@
...
-  width: 100%;
-}
-@media screen and (max-width: 640px) {
-  .lh-columns {
-    flex-wrap: wrap;
-
-  }
+.lh-metrics-container {
+  display: grid;
+  grid-template-rows: 1fr 1fr 1fr;
```
Comment (paulirish) [inline lighthouse-core/report/html/report-styles.css:551]: dunno a lot about grid but does this need to be repeated 3 times?  what if another metric is added via custom config or w/e. do we need changes here?

File: lighthouse-core/report/html/report-styles.css [hunk 3]
```diff
@@ -544,50 +544,33 @@
...
-  margin-right: 24px;
+.lh-metric {
+  border-top: 1px solid var(--report-border-color-secondary);
 }
 
-@media screen and (max-width: 800px) {
-  .lh-column:first-of-type {
-    margin-right: 8px;
+@media screen and (min-width: 640px) {
+  .lh-metric:nth-last-child(-n+2) {
```
Comment (paulirish) [inline lighthouse-core/report/html/report-styles.css:561]: fancy footwork. 🕺

File: lighthouse-core/report/html/report-styles.css (hunk 2 above)
Comment (connorjclark) [reply lighthouse-core/report/html/report-styles.css:551]: right, a 7th metric would make a new column and be weird ...

if instead i used `grid-template-columns: 1fr 1fr` (and drop the `grid-flow: columns` so it defaults to `rows`) the order is no longer we want:

image

we forced this order before by using  two divs, but that didn't allow for alignment.

Comment (connorjclark) [reply lighthouse-core/report/html/report-styles.css:551]: idk how to solve this other than keeping as-is and being ok with this in for some reason there is another metric in the LHR 

image


I think we'd redesign this whole thing if we had >6 metrics here tho..

Comment (connorjclark) [reply lighthouse-core/report/html/report-styles.css:551]: another solution: interleave the creation of these `lh-metric` divs such that the rendering order is good. like `1 4 2 5 3 6` ...

File: lighthouse-core/report/html/renderer/performance-category-renderer.js (hunk 1 above)
Comment (connorjclark) [reply lighthouse-core/report/html/renderer/performance-category-renderer.js:141]: yeah lol


shouldnt we sort these by weight? the implicit order in the config is not sorted by weight so what we show in the first column is not necessarily the most "key" metrics


image

Comment (connorjclark) [reply lighthouse-core/report/html/renderer/performance-category-renderer.js]: (no–paul wants the order exactly like this, not ordered by weight)

File: lighthouse-core/report/html/report-styles.css (hunk 2 above)
Comment (connorjclark) [reply lighthouse-core/report/html/report-styles.css:551]: in other words, css grid doesnt allow for letting elements fill up a column 1 first, column 2 second, and then not create a 3rd column (what would that css even look like...)

File: lighthouse-core/report/html/renderer/performance-category-renderer.js (hunk 1 above)
Comment (paulirish) [reply lighthouse-core/report/html/renderer/performance-category-renderer.js]: nah i want first col to be FCP, SI, LCP. and second column TTI, TBT, CLS.

the hidden motivation is showing the timing metrics chronological. and the non timing metrics at the end.

File: lighthouse-core/report/html/report-styles.css (hunk 2 above)
Comment (paulirish) [reply lighthouse-core/report/html/report-styles.css:551]: gotcha. IMO with 7 metrics it doesnt look that bad. 

I'm fine with keeping the css as you have it.

Comment (connorjclark) [reply lighthouse-core/report/html/report-styles.css:551]: cool :beans:

**Pull request merged**